lib/flight_scheduler/message_processor.rb

```diff
@@ -40,10 +40,17 @@ def call(message)
       case command
 
       when 'JOB_ALLOCATED'
-        job_id, script, arguments = message[:job_id], message[:script], message[:arguments]
+        job_id    = message[:job_id]
+        script    = message[:script]
+        arguments = message[:arguments]
+        # Ensure env is indeed a hash otherwise all sorts of weirdness will
+        # break loose in Process.spawn
+        env       = message[:environment].map { |k, v| [k.to_s, v&.to_s] }.to_h
+
         Async.logger.info("Running job:#{job_id} script:#{script} arguments:#{arguments}")
+        Async.logger.debug("Environment: #{env.map { |k, v| "#{k}=#{v}" }.join("\n")}")
         begin
-          task = FlightScheduler::JobRunner.run_job(job_id, script, *arguments)
+          task = FlightScheduler::JobRunner.run_job(job_id, env, script, *arguments, unsetenv_others: true)
```

Comment:
I think it would be worth updating the signature of `run_job` to `run_job(job_id, env, *arguments, **options)`.  I was at first confused by how `env` was getting passed in.

Perhaps similar should happen  for `script` too.  Or perhaps not merging script with arguments seems more natural than merging an environment hash.